src/modules/samples/rust/src/lib.rs

```diff
@@ -14,6 +17,46 @@ pub const MMI_OK: i32 = 0;
 pub type MmiHandle = *mut c_void;
 pub type MmiJsonString = *mut c_char;
 
+#[derive(Debug)]
+enum MmiError {
+    FailedRead(Utf8Error),
+    Code(i32),
+    FailedAllocate(NulError),
+}
+impl From<Utf8Error> for MmiError {
```

Comment:
Is there a more generic trait implementation that will work for converting from error?

Reply:
What will happen if a new Error type needs to be captured? What does the process look like for a developer to integrate that into `MmiError`?

Reply:
Discussed offline, options were box<Error>, struct, or this. Liked this for simplicity and that the other options may not actually simplify.